engine/src/main/junit-shared/org/operaton/bpm/engine/test/util/JobExecutorHelper.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.operaton.bpm.engine.test.util;
+
+import org.operaton.bpm.BpmPlatform;
+import org.operaton.bpm.ProcessEngineService;
+import org.operaton.bpm.engine.ManagementService;
+import org.operaton.bpm.engine.ProcessEngine;
+import org.operaton.bpm.engine.ProcessEngineException;
+import org.operaton.bpm.engine.impl.ProcessEngineImpl;
+import org.operaton.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.operaton.bpm.engine.impl.jobexecutor.JobExecutor;
+import org.operaton.bpm.engine.impl.jobexecutor.ThreadPoolJobExecutor;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+
+/**
+ * Utility class for managing and ensuring the processing of jobs by the JobExecutor
+ * in a BPM execution environment. This class provides methods for waiting until
+ * all jobs are processed, querying the availability of jobs, and waiting for jobs
+ * running in the thread pool to finish execution.
+ * The methods offered by this class facilitate coordination of job processing
+ * during tasks such as testing, ensuring proper job execution, and system
+ * synchronization.
+ * Note that this class includes mechanisms for controlling the active state of the
+ * JobExecutor and employs configurable time intervals and timeouts for various
+ * operations. All methods are statically available since the class is not intended
+ * for instantiation.
+ */
+public class JobExecutorHelper {
+
+  public static final long CHECK_INTERVAL_MS = 500L;
+  public static final long JOBS_WAIT_TIMEOUT_MS = 20_000L;
```

Comment:
avoid making these 2 constants defaults public. Rather offer and use a method without these parameters.
Maybe use DEFAULT_ prefix?